### PR TITLE
Add WithComponentsConstraints to RFC 3739

### PR DESCRIPTION
**Pull request by russhousley**
Add WithComponentsConstraints to enforce the requirements that are indicated by comments in the ASN.1 module.

### CODE REVIEW

File: pyasn1_modules/rfc3739.py [hunk 1]
```diff
@@ -158,6 +160,12 @@ class SemanticsInformation(univ.Sequence):
         namedtype.OptionalNamedType('nameRegistrationAuthorities',
             NameRegistrationAuthorities())
     )
+    subtypeSpec = constraint.ConstraintsUnion(
```
Comment (etingof) [inline pyasn1_modules/rfc3739.py]: Would it make sense to have a unit test against a broken data structure i.e. w/o any of these components?

File: pyasn1_modules/rfc3739.py [hunk 2]
```diff
@@ -2,6 +2,8 @@
 # This file is part of pyasn1-modules software.
 #
 # Created by Russ Housley with assistance from asn1ate v.0.6.0.
+# Modified by Russ Housley to add WithComponentsConstraints to
+#   enforce the requirements that are indicated in comments.
```
Comment (etingof) [inline pyasn1_modules/rfc3739.py]: nit: stray indentation

Comment (russhousley) [reply pyasn1_modules/rfc3739.py]: This spacing seems right to me, but that is a matter of taste.

File: pyasn1_modules/rfc3739.py (hunk 1 above)
Comment (russhousley) [reply pyasn1_modules/rfc3739.py]: I can work on a unit test to test for a failure.

Comment (etingof) [reply pyasn1_modules/rfc3739.py]: Would you like to work on that within this PR or some other?

Comment (russhousley) [reply pyasn1_modules/rfc3739.py]: I will add it to the PR shortly

Comment (russhousley) [reply pyasn1_modules/rfc3739.py]: The updated test_rfc3739.py has been added.

File: tests/test_rfc3739.py [hunk 3]
```diff
@@ -111,6 +112,19 @@ class SequenceOfOID(univ.SequenceOf):
...
 
+class WithComponentsTestCase(unittest.TestCase):
+
+    def testDerCodec(self):
+
+        try:
+            si = rfc3739.SemanticsInformation()
+            substrate = der_encoder(si)
+            encode_without_error = True
+        except error.PyAsn1Error:
```
Comment (etingof) [inline tests/test_rfc3739.py]: There is an idiom for this in unittest:

    self.assertRaises(error.PyAsn1Error, der_encode, si)

effectively replacing the whole try/except block.

Does it make sense?

Comment (russhousley) [reply tests/test_rfc3739.py]: I did not know about assertRaises().  Nice improvement!

Comment (russhousley) [reply tests/test_rfc3739.py]: I just added the improved test_rfc3739.py to the PR

**Comment by etingof** [comment]
Thank you!

**Pull request merged**